### PR TITLE
Improvement: Add features to insert_df_to_hive_table()

### PR DESCRIPTION
**Pull request by dombean**
## Description

Modified `insert_df_to_hive_table` function in `cdp/io/output.py`. Added support for creating non-existent Hive tables, repartitioning by column or partition count, and handling missing columns with explicit type casting.

*You can delete options that are not relevant.*

- [x] New feature - *non-breaking change*

## Checklist:

- [x] I have performed a self-review of my own code.
- [x] I have commented my code appropriately, focusing on explaining my design decisions (explain why, not how).
- [x] I have made corresponding changes to the documentation (comments, docstring, etc.. )
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have updated the change log.

<br>

#  Peer review
Any new code includes all the following:

- **Documentation**: docstrings, comments have been added/ updated.
- **Style guidelines**: New code conforms to the project's contribution guidelines.
- **Functionality**: The code works as expected, handles expected edge cases and exceptions are handled appropriately.
- **Complexity**: The code is not overly complex, logic has been split into appropriately sized functions, etc..
- **Test coverage**: Unit tests cover essential functions for a reasonable range of inputs and conditions. Added and existing tests pass on my machine.

### Review comments
Suggestions should be tailored to the code that you are reviewing. Provide context.
Be critical and clear, but not mean. Ask questions and set actions.
<details><summary>These might include:</summary>

- bugs that need fixing (does it work as expected? and does it work with other code
  that it is likely to interact with?)
- alternative methods (could it be written more efficiently or with more clarity?)
- documentation improvements (does the documentation reflect how the code actually works?)
- additional tests that should be implemented
  - Do the tests effectively assure that it
  works correctly? Are there additional edge cases/ negative tests to be considered?
- code style improvements (could the code be written more clearly?)
</details>
<br>

*Further reading: [code review best practices](https://best-practice-and-impact.github.io/qa-of-code-guidance/peer_review.html)*

### CODE REVIEW

File: rdsa_utils/cdp/io/output.py [hunk 1]
```diff
@@ -65,36 +92,77 @@ def insert_df_to_hive_table(
...
+    >>> insert_df_to_hive_table(
+    ...     spark=spark,
+    ...     df=df,
+    ...     table_name="my_database.my_table",
+    ...     overwrite=True
+    ... )
+
+    Write a DataFrame to a Hive table with missing columns filled:
+    >>> insert_df_to_hive_table(
+    ...     spark=spark,
```
Comment (cpkson) [inline rdsa_utils/cdp/io/output.py:119]: worth adding what the missing cols are filled with. I.E do you use a blank string for a string column, or a NULL etc. Are missing int columns filled with defaults of 0 or NaNs?

File: rdsa_utils/cdp/io/output.py [hunk 2]
```diff
@@ -65,36 +92,77 @@ def insert_df_to_hive_table(
...
+    ...     table_name="my_database.my_table",
+    ...     repartition_column="partition_column"
+    ... )
+
+    Repartition into a fixed number of partitions before writing:
+    >>> insert_df_to_hive_table(
+    ...     spark=spark,
+    ...     df=df,
+    ...     table_name="my_database.my_table",
+    ...     repartition_column=10
```
Comment (cpkson) [inline rdsa_utils/cdp/io/output.py:138]: I think it's worth changing the name of the `repartition_column ` variable. It doesn't make sense to be passing it an integer and a column string. I'd change it to `repartition_data_by`

So it looks like `insert_df_to_hive_table(
spark=spark,df=df,
table_name="my_database.my_table",
repartition_data_by=10)`

**Review by cpkson — CHANGES_REQUESTED** [review]
@dombean only some minor changes, then i'll test it :)

File: rdsa_utils/cdp/io/output.py (hunk 1 above)
Comment (dombean) [reply rdsa_utils/cdp/io/output.py:123]: Added to docstring regarding this and made the logic better for fill_missing_col, it will match df column to underlying Hive Schema
